susemanager/bin/mgr-setup

```diff
@@ -199,75 +199,108 @@ cleanup_hostname() {
-                exit 1
-            }
+        fi
+        systemctl start postgresql
+        if ! exists_db $MANAGER_DB_NAME; then
+            # required for postgresql <= 13 before creating the user
+            echo "password_encryption = 'scram-sha-256'" >> /var/lib/pgsql/data/postgresql.conf
+            systemctl restart postgresql
+            su - postgres -c "createdb -E UTF8 $MANAGER_DB_NAME ; echo \"CREATE ROLE $MANAGER_USER PASSWORD '$MANAGER_PASS' SUPERUSER NOCREATEDB NOCREATEROLE INHERIT LOGIN;\" | psql"
+
+            echo "local $MANAGER_DB_NAME postgres peer
+    local $MANAGER_DB_NAME $MANAGER_USER scram-sha-256
+    host $MANAGER_DB_NAME $MANAGER_USER 127.0.0.1/32 scram-sha-256
+    host $MANAGER_DB_NAME $MANAGER_USER ::1/128 scram-sha-256
+    " > /tmp/pg_hba.conf
+            cat /var/lib/pgsql/data/pg_hba.conf >> /tmp/pg_hba.conf
+            mv /var/lib/pgsql/data/pg_hba.conf /var/lib/pgsql/data/pg_hba.conf.bak
+            mv /tmp/pg_hba.conf /var/lib/pgsql/data/pg_hba.conf
+            chmod 600 /var/lib/pgsql/data/pg_hba.conf
+            chown postgres:postgres /var/lib/pgsql/data/pg_hba.conf
+            systemctl restart postgresql
+        else
+            echo "Database exists. Preparing for resetup. All data will be removed."
         fi
     else
-        # Define POSTGRES_LANG in system wide config. (Used during any DB creation)
-        . /etc/sysconfig/postgresql
-        if [ -z $POSTGRES_LANG ]; then
-            grep "^POSTGRES_LANG" /etc/sysconfig/postgresql > /dev/null 2>&1
-            if [ $? = 0 ]; then
-                sed -i -e "s/^POSTGRES_LANG.*$/POSTGRES_LANG=\"en_US.UTF-8\"/" /etc/sysconfig/postgresql
+        if ! exists_db $MANAGER_DB_NAME; then
+            echo "CREATE DATABASE $MANAGER_DB_NAME ENCODING = UTF8 ;" | ( export PGPASSWORD=$EXTERNAL_DB_PASS; exec psql -h $MANAGER_DB_HOST -U $EXTERNAL_DB_USER)
+        fi
+        if ! exists_user $MANAGER_USER; then
+            echo  $EXTERNAL_DB_PROVIDER
+            if [[ -n $EXTERNAL_DB_PROVIDER && "${EXTERNAL_DB_PROVIDER,,}" == "aws" ]] ; then
+                echo "CREATE ROLE $MANAGER_USER PASSWORD '$MANAGER_PASS'  NOCREATEDB NOCREATEROLE INHERIT LOGIN;" | ( export PGPASSWORD=$EXTERNAL_DB_PASS; exec psql -h $MANAGER_DB_HOST -U $EXTERNAL_DB_USER)
+                echo "GRANT rds_superuser to $MANAGER_USER;" | ( export PGPASSWORD=$EXTERNAL_DB_PASS; exec psql -h $MANAGER_DB_HOST -U $EXTERNAL_DB_USER)
+                echo "GRANT create on database $MANAGER_DB_NAME to $MANAGER_USER;" | ( export PGPASSWORD=$EXTERNAL_DB_PASS; exec psql -h $MANAGER_DB_HOST -U $EXTERNAL_DB_USER)
             else
-                echo "POSTGRES_LANG=\"en_US.UTF-8\"" >> /etc/sysconfig/postgresql
+                echo "CREATE ROLE $MANAGER_USER PASSWORD '$MANAGER_PASS' SUPERUSER NOCREATEDB NOCREATEROLE INHERIT LOGIN;" | ( export PGPASSWORD=$EXTERNAL_DB_PASS; exec psql -h $MANAGER_DB_HOST -U $EXTERNAL_DB_USER)
```

Comment:
Maybe we should have the user creation before the database creation and replace $EXTERNAL_DB_USER by $MANAGER_USER.

Reply:
I think we still need two users here. one (typically postgres) that is able to connect to the deployed remote postgres instance with the correct  role permissions that allows the creation of the susemanager database and the spacewalk role  (with SUPERUSER or the equivalent) .  perhaps I am not fully understanding your comment. Could you please provide more details?